src/firebase-resource-manager.ts

```diff
@@ -0,0 +1,119 @@
+import { OperationPollerOptions, pollOperation } from "./operation-poller";
+import { OraWrapper } from "./oraWrapper";
+
+const ONE_SECOND_MILLIS = 1000;
+
+export enum ParentResourceType {
+  ORGANIZATION = "organization",
+  FOLDER = "folder",
+}
+
+export interface ParentResource {
+  id: string;
+  type: ParentResourceType;
+}
+
+export async function createFirebaseProject(
+  projectId: string,
+  projectDisplayName: string,
+  parentResource?: ParentResource
+): Promise<{ projectId: string }> {
+  await createCloudProject(projectId, projectDisplayName, parentResource);
+  const projectInfo = await addFirebaseToCloudProject(projectId);
+
+  logger.info("");
+  logger.info("🎉🎉🎉 Your Firebase project is ready! 🎉🎉🎉");
+  logger.info("");
+  logger.info("Project information:");
+  logger.info(`   - Project ID: ${clc.bold(projectInfo.projectId)}`);
+  logger.info(`   - Project Name: ${clc.bold(projectInfo.displayName)}`);
+  logger.info("");
+  logger.info("Firebase console is available at");
+  logger.info(`https://console.firebase.google.com/project/${clc.bold(projectId)}/overview`);
+  return { projectId };
+}
+
+/**
+ * Send an API request to create a new Google Cloud Platform project and poll the LRO to get the
+ * new project information.
+ * @return {Promise} this function returns a promise that resolves to the new cloud project
+ *     information
+ */
+async function createCloudProject(
+  projectId: string,
+  projectDisplayName: string,
+  parentResource?: ParentResource
+): Promise<any> {
+  const spinner = new OraWrapper("Creating Google Cloud Platform project");
+  spinner.start();
+
+  try {
+    const response = await api.request("POST", "/v1/projects", {
+      auth: true,
+      origin: api.resourceManagerOrigin,
+      timeout: 15 * ONE_SECOND_MILLIS,
+      data: { projectId, name: projectDisplayName, parent: parentResource },
+    });
+
+    const projectInfo = await pollOperation<any>({
+      pollerName: "Project Creation Poller",
+      apiOrigin: api.resourceManagerOrigin,
+      apiVersion: "v1",
+      operationResourceName: response.body.name /* LRO resource name */,
+    });
+    spinner.succeed();
+    return projectInfo;
+  } catch (err) {
+    spinner.fail();
+    logger.debug(err.message);
+    throw new FirebaseError(
+      "Failed to create Google Cloud project. See firebase-debug.log for more info.",
+      { exit: 2, original: err }
+    );
+  }
+}
+
+/**
+ * Send an API request to add Firebase to the Google Cloud Platform project and poll the LRO
+ * to get the new Firebase project information.
+ * @return {Promise} this function returns a promise that resolves to the new firebase project
+ *    information
+ */
+async function addFirebaseToCloudProject(projectId: string): Promise<any> {
+  const spinner = new OraWrapper("Adding Firebase to Google Cloud project");
+  spinner.start();
+
+  // TODO(caot): Removed when "Deferred Analytics" and "Deferred Location" are launched
```

Comment:
Leaving this comment open until this TODO is resolved

Reply:
I think this is launched now, right?

Reply:
Only deferred location is launched.

Reply:
Resolved TODO